Fix warnings during compilation time on ARM (32 bits)

thiagoftsm opened the pull request:
##### Summary
Fixes #13637 

While we were developing some PRs, we brought some warnings that appear only on specific platforms. This PR is fixing issues that were appearning on `ArmVL7`. For an update report, please, take a look at  [errors](https://github.com/netdata/netdata/files/9603869/original_err.txt)

##### Test Plan
1. Compile current master branch on `Raspberry PI`, and you will see a lot of warnings.
2. Now compile this branch and verify that all warnings are gone.

##### Additional Information
In both scenarios I used the flags `CFLAGS="-Og -ggdb -Wall -Wextra -fno-omit-frame-pointer -Wformat-signedness -fstack-protector-all -Wformat-truncation=2 -Wunused-result -DNETDATA_INTERNAL_CHECKS=1 -D_FORTIFY_SOURCE=2 -DNETDATA_VERIFY_LOCKS=1"` to compile and take a look in the warnings.

<details> <summary>For users: How does this change affect me?</summary>
Describe the PR affects users: 
- Which area of Netdata is affected by the change? Compilation
- Can they see the change or is it an under the hood? If they can see it, where? they cannot see.
- How is the user impacted by the change? During compilation time Netdata does not show warnings.
- What are there any benefits of the change? Less noise
</details>

MrZammler commented:
Hi Thiago! Tried it in x86 32bit, which the PR removes almost all warnings! Nice!

I only get these two:

```
database/sqlite/sqlite_aclk_chart.c: In function 'aclk_update_retention':
database/sqlite/sqlite_aclk_chart.c:931:46: warning: comparison of integer expressions of different signedness: 'time_t' {aka 'long int'} and 'uint32_t' {aka 'unsigned int'} [-Wsign-compare]
  931 |             int live = ((now - last_entry_t) < (RRDSET_MINIMUM_DIM_LIVE_MULTIPLIER * update_every));
      |                                              ^
  CC       database/sqlite/sqlite3.o
database/sqlite/sqlite_aclk_alert.c: In function 'have_recent_alarm':
database/sqlite/sqlite_aclk_alert.c:880:55: warning: comparison of integer expressions of different signedness: 'uint32_t' {aka 'unsigned int'} and 'time_t' {aka 'long int'} [-Wsign-compare]
  880 |         if (ae->alarm_id == alarm_id && ae->unique_id > mark &&
      |                                                       ^
```

These don't seem 32bit related... I believe in the second case we should change `mark` to be `uint32_t` in the function definition. You also have it in `health/health_json.c`, not sure either there if `mark` should be time_t.

The first case can be casted?

MrZammler commented:
Tested the build with this PR on x86 32, cloud connectivity, etc, no issues so far!

thiagoftsm commented:
Hello @MrZammler ,

Firstly sorry for the delay, but I was busy with our `go.d.plugin`.
I push a commit addressing your comments. I did not have these issues on my Raspberry, thanks to report! Considering that values will be always positive, I used cast for `uint32_t`. I decided to use the road with less changes.

@vlvkobal  I will review the code on Sunday to adjust for your suggestion. Thanks!

thiagoftsm commented on daemon/unit_test.c:
```diff
@@ -1636,28 +1636,28 @@ int unit_test_bitmap256(void) {
         fprintf(stderr, "%s() INDEX 0 is fully set OK\n", __FUNCTION__);
     else {
-        fprintf(stderr, "%s() INDEX 0 is %lx expected 0xffffffffffffffff\n", __FUNCTION__, test_bitmap.data[0]);
+        fprintf(stderr, "%s() INDEX 0 is %lx expected 0xffffffffffffffff\n", __FUNCTION__, (long unsigned int) test_bitmap.data[0]);
```
I accepted your suggestion and  updated all formats that have `uint64_t`. Thanks again!
